src/components/src/PurchaseOrderWidget/purchase.order.widget.tsx

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright © 2019 Elastic Path Software Inc. All rights reserved.
+ *
+ * This is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React from 'react';
+import intl from 'react-intl-universal';
+import { getConfig, IEpConfig } from '../utils/ConfigProvider';
+import './purchase.order.widget.less';
+import PurchaseOrderWidgetModal from '../PurchaseOrderWidgetModal/purchase.order.widget.modal';
+
+let Config: IEpConfig | any = {};
+
+interface PurchaseOrderWidgetState {
+  poModalOpen: boolean,
+  isChosen: boolean,
+}
+
+interface PurchaseOrderWidgetProps {
+  /** Called when the purchase order widget is closed -- meant as a callback to refresh the PO widget modal. */
+  onChange?: (...args: any[]) => any,
+  /** Order data to be passed to PO widget */
+  orderPaymentData: any,
+}
+
+class PurchaseOrderWidget extends React.Component<PurchaseOrderWidgetProps, PurchaseOrderWidgetState> {
+  constructor(props) {
+    super(props);
+    const epConfig = getConfig();
+    Config = epConfig.config;
+    this.state = {
+      isChosen: false,
+      poModalOpen: false,
+    };
+    this.renderPONumber = this.renderPONumber.bind(this);
+    this.doesPayWithPOMethodExist = this.doesPayWithPOMethodExist.bind(this);
+    this.getPOPaymentMethodUri = this.getPOPaymentMethodUri.bind(this);
+    this.getPOMethodInfoFromOrderData = this.getPOMethodInfoFromOrderData.bind(this);
+    this.renderModal = this.renderModal.bind(this);
+    this.handleCloseModal = this.handleCloseModal.bind(this);
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    const { orderPaymentData } = props;
+
+    const chosen = PurchaseOrderWidget.getChosenFromOrderData(orderPaymentData);
+    try {
+      const newPONumber = chosen[0]._description[0]['payment-instrument-identification-attributes']['purchase-order'];
+
+      if (newPONumber !== undefined) {
+        return { isChosen: true };
+      }
+    // eslint-disable-next-line no-empty
+    } catch (err) {}
```

Comment:
Have you encountered any error scenarios with these empty catch blocks? Does the UI respond with anything?